testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveBasicTableStatistics.java

```diff
@@ -403,13 +421,150 @@ public void testInsertBucketedPartitioned()
+                "   bucketed_by = ARRAY['n_nationkey'], " +
+                "   bucket_count = 10 " +
+                ")", tableName));
+
+        onTrino().executeQuery(format("INSERT INTO %s (n_nationkey, n_name, n_regionkey) VALUES (1, 'Finlandia', 1), (2, 'Utopia', 1)", tableName));
+
+        onTrino().executeQuery(format("ANALYZE %s", tableName));
+        BasicStatistics partitionStatisticsAfterAnalyze = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=1");
+        assertThatStatisticsArePresent(partitionStatisticsAfterAnalyze);
+
+        operation.accept(tableName);
+
+        BasicStatistics partitionStatisticsAfterOperation = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=1");
+        assertThatStatisticsArePresent(partitionStatisticsAfterOperation);
+
+        checkStatisticsChange(statisticsChange, partitionStatisticsAfterOperation, partitionStatisticsAfterAnalyze);
+
+        onTrino().executeQuery(format("ANALYZE %s", tableName));
+
+        BasicStatistics statisticsAfterLastAnalyze = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=1");
+
+        checkStatisticsChange(statisticsChange, statisticsAfterLastAnalyze, partitionStatisticsAfterAnalyze);
+    }
+
+    private void checkStatisticsChange(StatisticsChange statisticsChange, BasicStatistics statisticsAfterOperation, BasicStatistics statisticsAfterAnalyze)
+    {
+        switch (statisticsChange) {
+            case UNCHANGED:
+                assertThat(statisticsAfterOperation.getNumRows().getAsLong()).isEqualTo(statisticsAfterAnalyze.getNumRows().getAsLong());
+                assertThat(statisticsAfterOperation.getNumFiles().getAsLong()).isGreaterThanOrEqualTo(statisticsAfterAnalyze.getNumFiles().getAsLong());
+                assertThat(statisticsAfterOperation.getTotalSize().getAsLong()).isGreaterThanOrEqualTo(statisticsAfterAnalyze.getTotalSize().getAsLong());
+                assertThat(statisticsAfterOperation.getRawDataSize().getAsLong()).isGreaterThanOrEqualTo(statisticsAfterAnalyze.getRawDataSize().getAsLong());
+                break;
+            case INCREASED:
+                assertThat(statisticsAfterOperation.getNumRows().getAsLong()).isGreaterThanOrEqualTo(statisticsAfterAnalyze.getNumRows().getAsLong());
+                assertThat(statisticsAfterOperation.getNumFiles().getAsLong()).isGreaterThanOrEqualTo(statisticsAfterAnalyze.getNumFiles().getAsLong());
+                assertThat(statisticsAfterOperation.getTotalSize().getAsLong()).isGreaterThanOrEqualTo(statisticsAfterAnalyze.getTotalSize().getAsLong());
+                assertThat(statisticsAfterOperation.getRawDataSize().getAsLong()).isGreaterThanOrEqualTo(statisticsAfterAnalyze.getRawDataSize().getAsLong());
+                break;
+            case DECREASED:
+                assertThat(statisticsAfterOperation.getNumRows().getAsLong()).isLessThanOrEqualTo(statisticsAfterAnalyze.getNumRows().getAsLong());
```

Comment:
Yes; Hive transactional DELETE adds deletion files in order to remove rows, so the totalSize and rawDataSize in general increase rather than decreasing.  I added code to check that they were the same or increased.